WordPressKitTests/Utilities/URLSessionHelperTests.swift

```diff
@@ -0,0 +1,130 @@
+                return result
+            } failure: { _ in
+                // Do nothing
+                return nil
+            }
+        await fulfillment(of: [expectation])
+    }
+
+    func testUnacceptable500() async throws {
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(data: "Internal server error".data(using: .utf8)!, statusCode: 500, headers: nil)
+        }
+
+        let result = await URLSession.shared.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
+
+        // The result is a successful result. This line should not throw
+        _ = try result.get()
+
+        let expectation = expectation(description: "API call returns server error")
+        _ = result
+            .assessStatusCode { result in
+                return result
+            } failure: { result in
+                XCTAssertEqual(String(data: result.body, encoding: .utf8), "Internal server error")
+                expectation.fulfill()
+                return nil
+            }
+        await fulfillment(of: [expectation])
+    }
+
+    func testAcceptable404() async throws {
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(data: "Not found".data(using: .utf8)!, statusCode: 404, headers: nil)
+        }
+
+        let result = await URLSession.shared.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
+
+        // The result is a successful result. This line should not throw
+        _ = try result.get()
+
+        let expectation = expectation(description: "API call returns not found")
+        _ = result
+            .assessStatusCode(acceptable: [200...299, 400...499]) { result in
+                XCTAssertEqual(String(data: result.body, encoding: .utf8), "Not found")
+                expectation.fulfill()
+                return result
+            } failure: { result in
+                return nil
+            }
+        await fulfillment(of: [expectation])
+    }
+
+    func testParseError() async throws {
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(data: "Not found".data(using: .utf8)!, statusCode: 404, headers: nil)
+        }
+
+        let result = await URLSession.shared.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
+
+        // The result is a successful result. This line should not throw
+        _ = try result.get()
+
+        let expectation = expectation(description: "API call returns not found")
+        let parsedResult = result
+            .assessStatusCode { result in
+                return result
+            } failure: { result in
+                expectation.fulfill()
+                if result.response.statusCode == 404 {
+                    return .postNotFound
+                }
+                return nil
+            }
+        await fulfillment(of: [expectation])
```

Comment:
If I read this correctly based also on the info in the PR description, these tests are previews of how we'd use the code in real life to make API requests.

With that in mind, what do you think of using a name like map?

```swift
let session = ...

let result = await session.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
   .map(
     acceptableStatusCodes: [200...299],
     success: { result in ... }
     failure: { result in ... }
  )
```

With default parameter:

```swift
let session = ...

let result = await session.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
   .map(
     success: { result in ... }
     failure: { result in ... }
  )
```

I'm suggesting this because "assess" doesn't make me think of an operation that would apply the given transformations to `self` and return the result. 🤷‍♂️